core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -337,8 +341,42 @@ public void seek(long pos) throws IOException {
     }
     if (pos < mPos) {
       mEOF = false;
+      if (mDataReader instanceof NaiveSharedGrpcDataReader) {
+        NaiveSharedGrpcDataReader reader = (NaiveSharedGrpcDataReader)mDataReader;
+        reader.seek(pos);
+        if (mCurrentChunk != null) {
+          mCurrentChunk.release();
+          mCurrentChunk = null;
+        }
+      } else {
+        closeDataReader();
+      }
+    } else if (pos < mLength) {
+      // Try to read data already in queue
```

Comment:
Is the overhead introduced by closing and creating a data reader consistently bigger than the overhead of reading all data already in queue? 
Is the seek(pos) in your case (fuse scenario) always jump for a small step forward? If seek jump for a small step forward, read data already in the queue is helpful, but if seek jump for a large step or jump backward, may not worth reading data in the queue.
@apc999 in other Alluxio read scenarios (other than fuse scenario), seek() will jump a small step forward or all other cases have a large chance to happen as well?